packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts

```diff
@@ -259,10 +260,46 @@ export class PureDataObjectFactory<
 		return this.createNonRootInstanceCore(runtime, [this.type], initialState);
 	}
 
+	/**
+	 * Creates a new instance of the object.
+	 * @param runtime - container runtime. It is the runtime that will be used to create the object. It will produce
+	 * the underlying infrastructure to get the data object to operate.
+	 * @param initialState - The initial state to provide to the created component.
+	 * @param packagePath - The path to the data store factory to use to create the data object.
+	 * @returns an array containing the object created by this factory and an IDataStore object that enables users to
+	 * alias the data object.
+	 * The data object is attached only when it is attached to the handle graph that connects to an aliased object or
+	 * when the data object is aliased.
+	 */
+	public async constructInstance(
+		containerRuntime: IContainerRuntimeBase,
+		initialState?: I["InitialState"],
+		packagePath?: Readonly<string[]>,
+	): Promise<[TObj, IDataStore]> {
+		const context = containerRuntime.createDetachedDataStore(packagePath ?? [this.type]);
+		const { instance, runtime } = await createDataObject(
+			this.ctor,
+			context,
+			this.sharedObjectRegistry,
+			this.optionalProviders,
+			this.runtimeClass,
+			false, // existing
+			initialState,
+		);
+		const dataStore = await context.attachRuntime(this, runtime);
+
+		return [instance, dataStore];
+	}
+
 	/**
 	 * Creates a new root instance of the object. Uses container's registry to find this factory.
 	 * It's expected that only container owners would use this functionality, as only such developers
 	 * have knowledge of entries in container registry.
+	 *
+	 * This immediately makes the root instance live. If the container is attached before this is called, changes will
```

Comment:
Adding this comment seems unnecessary and even confusing. I am not sure we use the term live consistently.
Also, the concepts of what happened in detached vs attached containers is more widely applied doesn't need to be called out here explicitly.